js/fcoin.js

```diff
@@ -0,0 +1,572 @@
+                },
+            },
+        });
+    }
+
+    async fetchMarkets () {
+        let res = await this.publicGetSymbols ();
+        let result = [];
+        if (res['status'] === 0) {
+            let markets = res['data'];
+            for (let p = 0; p < markets.length; p++) {
+                let market = markets[p];
+                let id = market['name'];
+                let baseId = market['base_currency'].toUpperCase ();
+                let quoteId = market['quote_currency'].toUpperCase ();
+                let base = this.commonCurrencyCode (baseId);
+                let quote = this.commonCurrencyCode (quoteId);
+                let symbol = base + '/' + quote;
+                let precision = {
+                    'price': market['price_decimal'],
+                    'amount': market['amount_decimal'],
+                };
+                let limits = {
+                    'price': {
+                        'min': Math.pow (10, -precision['price']),
+                        'max': Math.pow (10, precision['price']),
+                    },
+                };
+                if (symbol in this.limits) {
+                    limits['amount'] = {
+                        'min': this.safeFloat (this.limits[symbol]['amount'], 'min'),
+                        'max': this.safeFloat (this.limits[symbol]['amount'], 'max'),
+                    };
+                } else {
+                    limits['amount'] = {
+                        'min': 0.01,
+                        'max': 100000,
+                    };
+                }
+                result.push ({
+                    'id': id,
+                    'symbol': symbol,
+                    'base': base,
+                    'quote': quote,
+                    'baseId': baseId,
+                    'quoteId': quoteId,
+                    'active': true,
+                    'precision': precision,
+                    'limits': limits,
+                    'info': market,
+                });
+            }
+        }
+        return result;
+    }
+
+    async fetchBalance (params = {}) {
+        await this.loadMarkets ();
+        let res = await this.privateGetAccountsBalance ();
+        let result = { 'info': res };
+        let balances = res['data'];
+        for (let i = 0; i < balances.length; i++) {
+            let balance = balances[i];
+            let currency = balance['currency'];
+            let uppercase = currency.toUpperCase ();
+            uppercase = this.commonCurrencyCode (uppercase);
+            let account = this.account ();
+            account['free'] = parseFloat (balance['available']);
+            account['total'] = parseFloat (balance['balance']);
+            account['used'] = parseFloat (balance['frozen']);
+            result[uppercase] = account;
+        }
+        return this.parseBalance (result);
+    }
+
+    convertOrderBook (bidasks) {
+        let newbidasks = [];
+        let length = bidasks.length / 2;
```

Comment:
This line won't transpile properly, because we inly use `.length` for string lengths, not for array lengths. To get the array length the proper transpileable syntax is:

```
let length = bidasks.length; // this way it will work (it has to end with `length;`
let halfLengh = length / 2;
```